cloudify_awssdk/ec2/resources/instances.py

```diff
@@ -412,3 +432,37 @@ def _handle_userdata(existing_userdata):
             [existing_userdata, install_agent_userdata])
 
     return final_userdata
+
+
+def _handle_password(iface):
+    if not ctx.node.properties['use_password']:
+        return True
+    key_data = ctx.node.properties['agent_config'].get('key')
```

Comment:
Are we sure that `agent_config` will be always provided ?

Reply:
nope. that's why we check for a key pair attached via a relationship.

Reply:
👍